src/FTPWindow.cpp

```diff
@@ -844,27 +843,20 @@ void TFTPWindow::Mkdir()
 
 void TFTPWindow::CopyUrl()
 {
-	BRow *row = fRemoteFileView->CurrentSelection();
+	BRow* row = fRemoteFileView->CurrentSelection();
 	if (row == NULL) return;
 
-	const char *intName
-		= ((BStringField *)row->GetField(CLM_INTERNAL_NAME))->String();
-
 	BString url;
-	if (fCurrentRemoteDir.Length() > 0 &&
-		fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] == '/') {
-
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << intName;
-	} else {
-		url << "ftp://" << fHost << ":" << fPort
-			<< fCurrentRemoteDir << "/" << intName;
-	}
+	url << "ftp://" << fHost << ":" << fPort << fCurrentRemoteDir;
+	if (fCurrentRemoteDir.Length() <= 0
+		|| fCurrentRemoteDir[fCurrentRemoteDir.Length() - 1] != '/')
+		url << "/";
+	url << ((BStringField *)row->GetField(CLM_INTERNAL_NAME))->String();
 
-	BMessage *clip = (BMessage *)NULL;
 	if (be_clipboard->Lock()) {
 		be_clipboard->Clear();
-		if (clip = be_clipboard->Data()) {
+		BMessage* clip = be_clipboard->Data();
+		if (clip == B_OK) {
```

Comment:
```suggestion
		if (clip != NULL) {
```

Reply:
Oh, silly me! I thought of it as if it returned with a status_t. Any way I fixed it.